F50/Platform/Platform.android.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Android.App;
+using Android.Content;
+using Android.OS;
+
+namespace Xamarin.F50
+{
+	public partial class Platform
+	{
+		static Application application;
+		static ActivityLifecycleContextListener lifecycleListener;
+
+		internal static Context CurrentContext =>
+			lifecycleListener?.Context ?? application?.ApplicationContext;
+
+		internal static Activity CurrentActivity =>
+			lifecycleListener?.Activity;
+
+		public static void Init(Activity activity, Bundle bundle)
+		{
+			lifecycleListener = new ActivityLifecycleContextListener();
+			application = activity.Application;
+			application.RegisterActivityLifecycleCallbacks(lifecycleListener);
+		}
+	}
+
+	class ActivityLifecycleContextListener : Java.Lang.Object, Application.IActivityLifecycleCallbacks
+	{
+		Activity currentActivity = null;

```

Comment:
Should this be a `WeakReference` at all? What are the implications of holding a reference to an object. I know it is in the foreground at this point, but we are effectively keeping a static reference to an object.